compiler/front/optionsprocessor.nim

```diff
@@ -0,0 +1,1836 @@
+    # `foreignPackageNotes` so that they can enable
+    # `hints|warnings|warningAsErrors` for all the code they depend on.
+    return ProcessNoteResult(kind: procNoteExpectedOnOrOff, switch: arg, argVal: val)
+  else:
+    let isOn = val == "on"
+    if isOn and id.normalize == "all":
+      return ProcessNoteResult(kind: procNoteOnlyAllOffSupported, switch: arg, argVal: val)
+
+    for n in notes:
+      if n notin conf.cmdlineNotes or pass == passCmd1:
+        if pass == passCmd1:
+          conf.incl(cnCmdline, n)
+
+        conf.incl(cnModifiedy, n)
+
+        if state in {wWarningAsError, wHintAsError}:
+          conf.flip(cnWarnAsError, n, isOn)
+        else:
+          conf.flip(cnCurrent, n, isOn)
+          conf.flip(cnMainPackage, n, isOn)
+
+        if not isOn:
+          conf.excl(cnForeign, n)
+  
+  result = ProcessNoteResult(kind: procNoteSuccess)
+
+proc processCompile(conf: ConfigRef; filename: string) =
+  # TODO: inline
+  var found = findFile(conf, filename)
+  if found.isEmpty: found = AbsoluteFile filename
+  extccomp.addExternalFileToCompile(conf, found)
+
+# List of GC names for the error generation. It cannot be created
+# from enum set using `getEnumNames` because nim cmdline has
+# multiple names for the same garbage collector.
+const
+  gcNames = @[
+    "boehm", "refc", "markandsweep", "destructors", "arc", "orc",
+    "hooks", "go", "none", "stack", "regions",]
+
+  cmdNames = @[
+    "c", "cc", "compile", "compiletoc",
+    "compiletooc", "js", "compiletojs", "r", "run", "check", "e",
+    "doc2", "doc", "doc2tex", "rst2html", "rst2tex", "jsondoc2",
+    "jsondoc", "ctags", "buildindex", "gendepend", "dump", "parse", "rod",
+    "secret", "nop", "help", "jsonscript",]
+
+type
+  CompileOptArgCheckResult* = enum
+    compileOptArgCheckSuccessTrue
+    compileOptArgCheckSuccessFalse
+    compileOptArgCheckWarnFalseDeprecated
+    compileOptArgCheckFailedWithInvalidOption
+    compileOptArgCheckFailedWithUnexpectedValue
+
+func allowedCompileOptionsArgs*(switch: CmdSwitchKind): seq[string] =
+  # xxx: make this exhaustive somehow
+  case switch
+  of cmdSwitchBackend     : validBackends.toSeq.mapIt($it)
+  of cmdSwitchGc          : gcNames
+  of cmdSwitchDebugger    : @["native", "gdb", "on", "off"]
+  of cmdSwitchOpt         : @["speed", "size", "none"]
+  of cmdSwitchApp         : @["gui", "console", "lib", "staticlib"]
+  of cmdSwitchOs          : platform.listOSnames()
+  of cmdSwitchCpu         : platform.listCPUnames()
+  of cmdSwitchVerbosity   : @["0", "1", "2", "3"]
+  of cmdSwitchIncremental : @["on", "off", "writeonly", "readonly", "v2", "stress"]
+  of cmdSwitchCc          : listCCnames()
+  of cmdSwitchFilenames   : @["abs", "canonical", "legacyRelProj"]
+  of cmdSwitchProcessing  : @["dots", "filenames", "off"]
+  of cmdSwitchExperimental: experimentalFeatures.toSeq.mapIt($it)
+  of cmdSwitchExceptions  : @["native", "setjmp", "quirky", "goto"]
```

Comment:
```suggestion
  of cmdSwitchExceptions  : @["native", "goto"]
```

Reply:
Thanks, I thought I had gotten the last of these.